theseus_gui/src/pages/Settings.vue

```diff
@@ -1,7 +1,319 @@
-<script setup></script>
+<script setup>
+import { ref, watch } from 'vue'
+import { Card, Slider, DropdownSelect, Button, SearchIcon, PlayIcon } from 'omorphia'
+import { BrowseIcon } from '@/assets/icons'
+import { useTheming } from '@/store/state'
+import { get, set, deepEqual } from '@/helpers/settings'
+
+// Original object to track changes
+const originalSettings = ref(await get())
+// Object to bind
+const settings = ref(await get())
+
+const saveButton = ref(null)
+
+if (!settings.value.java_globals.JAVA_8)
+  settings.value.java_globals.JAVA_8 = { path: '', version: '' }
+if (!settings.value.java_globals.JAVA_17)
+  settings.value.java_globals.JAVA_17 = { path: '', version: '' }
+
+watch(settings.value, (newSettings) => {
+  // Validate the changed state
+  if (newSettings.hooks.pre_launch === '') delete settings.value.hooks.pre_launch
+  if (newSettings.hooks.wrapper === '') delete settings.value.hooks.wrapper
+  if (newSettings.hooks.post_exit === '') delete settings.value.hooks.post_exit
+
+  settings.value.max_concurrent_downloads = parseInt(newSettings.max_concurrent_downloads)
+
+  if (deepEqual(originalSettings.value, settings.value)) saveButton.value.$el.style.opacity = 0
+  else saveButton.value.$el.style.opacity = 1
```

Comment:
like I mentioned earlier this version of a settings button doesn't work with out styling